### ISSUE TITLE
QA effort for Mapbox SDK upgrade

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#3586]
This issue to track the QA effort to confirm functionality is maintained following mapbox upgrade. 
## Product/Feature Summary
The Mapbox version has been upgraded and we need to ensure that basic functionality has been maintained. We will also want to check to see if previously reported accessibility and usability issues have been resolved through this upgrade. 

## Proposed Scenarios
_Note: Facility Locator is part of the unauthenticated experience so no particular personas are needed for testing._

**Basic search scenarios:** 
For each of the following, the user should be able to view a map corresponding to specified location which contains icons marking the facilities returned in the search results. 
- User enables browser location, selects All facilities and performs a search. 
- User enters city and selects VA Health as Facility type. 
- User enters state and selects VA Health as Facility type. 
- User enters zip code and VA Health as Facility type. 
- User enters location and selects Benefits as Facility type. 
- User enters location and selects Cemetery as Facility type. 
- User enters location and selects Vet Center as Facility type. 

**Map functionality**
- User can view a hover over containing a description of the facility marked with the icon. 
- Clicking on the pin in the map connects the user to a detail page for that facility, rendered in the appropriate VA.gov page template.
   - The detail page will contain a thumbnail map of the selected facility. 
- User can interact with the window (ie, shrinking the window vertically) without a loss in functionality.  
- User can zoom to focus on either the details of a particular region without a loss in functionality. 
- User can zoom out to gain a quick overview of a wider area without a loss in functionality. 
- Only one request to the facilities/va endpoint should be made, with the box parameters corresponding to the bounds of the visible map and the address parameter corresponding to an address that is roughly at the center of the visible map. [#3206]

**Accessibility scenarios**
- The map is accessible to keyboard users. 
- Maps have no keyboard traps, including when in different views
- The individual map icons should announce the facility or provider name on keyboard focus.
- As a screen reader user, I want to hear relevant information when the map receives keyboard focus.
- Users can increase the size of the map, legend and any text without loss in functionality.
- Map renders in acceptable high contrast color.
- User is given a long description (which is equivalent to the map) for complex map images or image map areas and the description (or link) is located near the map. 
- Changes occur only after focus (e.g. not only on mouse hover).

**Keyboard Navigation (manual)**
[ ] Ensure all pages can be navigated with keyboard.
[ ] Ensure no links, buttons, or form elements are unreachable using Tab, Shift + Tab
[ ] Ensure custom rich components can be navigated using keyboard, arrow keys
[ ] Skip navigation link should be on all pages with navigation, large numbers of links
[ ] Ensure focusable elements (links, form inputs, buttons, radios, checkboxes) are all reachable by keyboard. Ensure any elements with a tabIndex="0" can be focused in the normal document flow.
[ ] Whenever possible, use proper semantic elements. For instance, it is better to use a <button> than re-create events and focus behaviors using custom tags.
[ ] Avoid keyboard traps. These are situations where users can tab into an interface, but can't tab out of it by pressing TAB or SHIFT+TAB.
[ ] Ensure your application has a predictable tabbing order. In left-right languages like English, this is assumed to be left to right, top to bottom. In multi-column layouts, use your best judgment. Interfaces that include a left navigation bar would likely allow users to focus each link or button in the left nav, then move focus to the main content area.
[ ] Offer ways to skip large groups of links like a skip to content link
[ ] Review the WebAIM keyboard accessibility guide for keyboard navigation patterns.

**Screen Readers (manual)**
[ ] User flow must be possible using a keyboard only (no pointing device). See the WebAIM keyboard accessibility guide for keyboard strokes.

**Color Tests**
[ ] Formation design system color palette has been tested to ensure compliance and usability.
[ ] Check for colorblindness issues in live sites using the Chrome Colorblinding plugin

### Mobile
- iOS
  - iPhone 11
  - iPhone 6
  - iPad 6th-gen
  - iPad Air 2
- Android
  - Galaxy S9
  - Galaxy S5

### Desktop
- Mac
  - Chrome
  - Firefox
  - Safari
- Windows
  - Chrome
  - Firefox
  - Edge
  - IE11

## SMEs

PM: Michelle Middaugh (@mmiddaugh)
UX: Aricka Lewis (@arickalewis1)
FE: Gibran Rodriguez (@brangi)
BE: Lance Sanchez (@darkbushido)

## Review

[Describe notes and comments derived from the review of QA activities with developers and product manager.]

## Staging Test-Readiness

No Staging test-accounts or test-data required -- Facility Locator is an unauthenticated experience.

## Definition of Done

[ ] - Engineering-team review completed.

[ ] - Test Plan finalized.

[ ] - QA-request ticket\* linked \[ticket-template coming soon]

**Comment by mmiddaugh** [comment]
@jenstrickland will it make sense to conduct a round of compliance testing on the map functionality once Gibran completes this upgrade? 

Goals: 
- Retest to confirm if previously reported issues are still present following the upgrade
- Check for the presence of new issues following upgrade

**Comment by jenstrickland** [comment]
@mmiddaugh - yes, recommend requesting a spot check accessibility review ahead of the Pre-launch 508 Review: [Schedule an accessibility review](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/.github/ISSUE_TEMPLATE/request_dev_review_vsa.md)

**Comment by mmiddaugh** [comment]
@tlei123 and @brangi will work together to create specific test scenarios related to areas of breaking change once those areas are identified.

**Comment by mmiddaugh** [comment]
This effort should also recheck [#2701], [#3206], [#3530]

**Comment by tlei123** [comment]
@brangi , is this PR the one that needs to be merged first before the Mapbox Upgrade is ready\* for QA?
https://github.com/department-of-veterans-affairs/vets-website/pull/11358

*I also recall in our planing mtg that we have some kind of API/backend dependency too.

**Comment by tlei123** [comment]
@mmiddaugh , for your review, here's the [Test Plan](https://dsvavsp.testrail.io/index.php?/plans/view/38) on TestRail.com (our online Test Case Management System (TCMS)).

I just emailed login credentials to you separately.

Please ping me on Slack when you're ready, and I'll walk you through TestRail.com.

Thanks much!

**Comment by brangi** [comment]
Hi @tlei123 ,

There is no backend dependency, all the changes are done in the front end application.

The PR has the conditional `if(environment.staging) { run new SDK version } else { run current  SDK version}` so that we can test it on staging once it is merged.

**Comment by tlei123** [comment]
@mmiddaugh @brangi , all test runs have now been completed and defects logged.

Test Plan Sumary Report is attached below, and also accessible online: https://dsvavsp.testrail.io/index.php?/reports/view/33
\[I'll email TestRail VSA shared-account credentials to you separately]

Defects logged:
* #4720 - Intermittent map-icon tap/click triggers blank page
* #4734 - Single-result searches fail to load map image on smartphones
* #4735 - Geolocation not auto-searching
* #4736 - City/State/Zip unfiltered search fails to load results
* #4764 - Map-icon displays incorrect location-popup if close to or overlapped by another icon

[vsa-facilities-mapbox-upgrade-test-plan-summary-report-200113.pdf](https://github.com/department-of-veterans-affairs/va.gov-team/files/4056838/vsa-facilities-mapbox-upgrade-test-plan-summary-report-200113.pdf)

**Comment by brangi** [comment]
Hi @tlei123,

thank you for your feedback. I can say with high confidence that the following defects are related and it's about geolocation not being detected:

#4734 - Single-result searches fail to load map image on smartphones
#4735 - Geolocation not auto-searching
#4736 - City/State/Zip unfiltered search fails to load results

For these ^ I will have a fix soon.

However defects:

#4720 - Intermittent map-icon tap/click triggers blank page
#4764 - Map-icon displays incorrect location-popup if close to or overlapped by another icon

Do not seem to be related to the sdk upgrade, @mmiddaugh any thoughts, should we fix all ?

**Comment by tlei123** [comment]
@mmiddaugh @brangi , I've just re-run all the failed/blocked test-cases, and here's my [Test Plan 2 Summary report on TestRail](https://dsvavsp.testrail.io/index.php?/reports/view/37) [PDF also attached here].  \[You should both already have received VSA shared-account credentials to log into TestRail, but feel free to ping me on Slack if you can't find them.]

#4736 (City/State/Zip unfiltered search fails to load results) is now fixed.

The other defects still persist, and I'll add a Comment to them with re-run result-details:
* #4720 (Intermittent map-icon tap/click triggers blank page)
* #4734 (Single-result searches fail to load map image on smartphones)
* #4735 (Geolocation not auto-searching)
* #4764 (Map-icon displays incorrect location-popup if close to or overlapped by another icon)

[mapbox-upgrade-test-plan-2-summary.pdf](https://github.com/department-of-veterans-affairs/va.gov-team/files/4100958/mapbox-upgrade-test-plan-2-summary.pdf)

**Comment by brangi** [comment]
@tlei123  As I mentioned above ^  the following are the only bugs introduced by the mapbox sdk upgrade (**main issue was auto detection of location**). This PR has fixed it - https://github.com/department-of-veterans-affairs/vets-website/pull/11498

#4734 - Single-result searches fail to load map image on smartphones
#4735 - Geolocation not auto-searching
#4736 - City/State/Zip unfiltered search fails to load results

How are you running the test cases? Are they Automated? Because I can confirm on staging.va.gov Gelocation + auto searching is working

**There might be some delay on acquiring the location at first (because the information is coming from mapbox backends) but it works,  see below:**

![autodetection](https://user-images.githubusercontent.com/100468/72952135-8f5dea80-3d56-11ea-91b7-166775d75bb7.gif)

**Comment by tlei123** [comment]
@brangi , I see @mmiddaugh has already Closed #4720 & #4764, so no problem there...

There's still real progress on your fixes...

#4734 is now only replicable on SauceLabs' emulators (Samsung S9, S6, & generic Android 8 emulator), so I added a Comment there about finding someone who has a real Samsung phone before further fixing.

#4735 is now replicable only on Mac Safari and SauceLabs-emulated Samsung S6 & generic Android 8.

The only real pressing issue is 4735 on Mac Safari -- @mmiddaugh can decide on what to do about Android devices (plus, whoever has a real device would need to do the QA).

Feel free to ping me tmrw (Thu) if you'd like me to screenshare what I saw.

**Comment by brangi** [comment]
> #4735 is now replicable only on Mac Safari and SauceLabs-emulated Samsung S6 & generic Android 8.
> 
> The only real pressing issue is 4735 on Mac Safari -- @mmiddaugh can decide on what to do about Android devices (plus, whoever has a real device would need to do the QA).

Works for me on Mac Safari just fine

![safariAutoLoc](https://user-images.githubusercontent.com/100468/72954490-2c705180-3d5e-11ea-99a5-9ca67034f4df.gif)

![safariAutoLocAsk](https://user-images.githubusercontent.com/100468/72954502-32fec900-3d5e-11ea-83dd-44422378a4bd.gif)

**Comment by mmiddaugh** [comment]
@tlei123 looks like we have a testing platform situation since @brangi is able to demonstrate it works on Mac Safari

**Comment by tlei123** [comment]
> @tlei123 looks like we have a testing platform situation since @brangi is able to demonstrate it works on Mac Safari

@mmiddaugh @brangi , my Mac Safari's loading geolocation auto-search today.  Not sure why I wasn't yesterday.  I just updated #4735.

Also, FYI, I found a Samsung owner who's gracious enough to help replicate on a real Samsung phone tmrw/Monday, so I'll have further feedback on this soon.
